criu/cr-service.c

```diff
@@ -737,6 +739,8 @@ static int dump_using_req(int sk, CriuOpts *req)
 
 	success = true;
 exit:
+	free_pidfd_store();
+pidfd_store_err:
```

Comment:
If init_pidfd_store_hash() fails you should also call free_pidfd_store() anyway.

Because init_pidfd_store_hash() can fail on some non-frist iteration in while on xmalloc when it already filled the hash with some number of entries..

Actually before this patchset we were calling free_pidfd_store always even

Reply:
Thanks, I didn't think of the xmalloc case before.

To avoid a redundant call to free_pidfd_store I can think of two ways:
1. I could add a ```free_pidfd_store``` if xmalloc fails and keep ```pidfd_store_err:``` label as it is.
2. I could remove ```free_pidfd_store``` from ```init_pidfd_store_hash``` and always call it after ```init_pidfd_store_hash```.

I am inclined towards the second option.

Reply:
I went the first option because I think ```init_pidfd_store_hash()``` should clean its failure, any feedback?